### PR TITLE
Tests sendwebpush

### PR DESCRIPTION
**Pull request by gauntface**
This adds methods for testing sendWebPush in it's current state.

### CODE REVIEW

**Comment by gauntface** [comment]
cc @petele @wibblymat @addyosmani @jeffposnick 

Anyone for a review?

File: gulp-tasks/test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
+  Copyright 2014 Google Inc. All Rights Reserved.
```
Comment (jeffposnick) [inline gulp-tasks/test.js]: Might as well switch to 2016.

File: gulp-tasks/test.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+/* eslint-env node */
+
+'use strict';
+
+const gulp = require('gulp');
+const mocha = require('gulp-mocha');
+
+gulp.task('test:manual', function() {
+  return gulp.src('./test/*.js', {read: false})
+    .pipe(mocha());
```
Comment (jeffposnick) [inline gulp-tasks/test.js]: I believe you'd want to add something like

``` js
.on('error', function(error) {
  console.error(error);
  process.exit(1);
});
```

to your Gulp stream to fail with a non-zero status if `gulp-mocha` fires an `error` event. That will, in turn, cause the `npm run test` to fail, which is what you'd want.

**Comment by jeffposnick** [comment]
Looks good in general. One piece of feedback with a scope that extends outside this PR is that it could be more maintainable to create a standalone file that included an `enum`-like mapping of error identifiers to error strings, and then refer to the errors via their identifier when constructing them in the library and when checking for them in the test.

**Comment by gauntface** [comment]
Ta for the review Jeff (The mocha exit with status 0 is super frustrating!!).

Raised an issue regarding the error message (agreed that this doesn't scale well or maintain well) https://github.com/GoogleChrome/push-encryption-node/issues/12

**Comment by gauntface** [comment]
@wibblymat when you get a chance, I'd appreciate you check over the code in push.js to make sure you are happy with them and I'm not missing anything.

**Comment by wibblymat** [comment]
Superceded by #18

**Pull request closed without merging**